src/main/kotlin/io/github/rybalkinsd/kohttp/interceptors/SigningInterceptor.kt

```diff
@@ -0,0 +1,23 @@
+package io.github.rybalkinsd.kohttp.interceptors
+
+import okhttp3.HttpUrl
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+
+class SigningInterceptor(private val parameterName: String, private val signer: HttpUrl.() -> String) : Interceptor {
```

Comment:
Need to refer common pattern:
LogInteceptor or LoggingInterceptor
Sign or Signing

Let’s define it in #64

Reply:
Can we provide any default for param name?

Reply:
Also need to add javadoc. We should mention if we guarantee any param order or not

Reply:
> Can we provide any default for param name?

Not sure. From what I saw all API's have different parameter names.

> Also need to add javadoc. We should mention if we guarantee any param order or not

Users have access to the `Url` object in the signer method, so I guess they have control over ordering.

Reply:
Yeah, looks like internal logic is straightforward. They use `ArrayList` and `add` method, like append to the end of list. Better to mention that it works this way in our class docs.

see `okhttp3.HttpUrl.Builder#addQueryParameter`